src/dynamic/HI_generator.cpp

```diff
@@ -42,32 +42,26 @@
 using TIME = float;
 
 // Ports for coupled models, we use the same in every level
-struct coupled_in_port : public cadmium::in_port<int>{};
-struct coupled_out_port : public cadmium::out_port<int>{};
-cadmium::dynamic::modeling::Ports coupled_in_ports = {typeid(coupled_in_port)};
-cadmium::dynamic::modeling::Ports coupled_out_ports = {typeid(coupled_out_port)};
-
-//A configured version of the devstone atomic, we use same configuration in every atomic.
-template<typename T>
-struct configured_atomic_devstone : devstone_atomic<T>{
-    configured_atomic_devstone(){
-        devstone_atomic<T>::period = 1;
-        devstone_atomic<T>::external_cycles = 100;
-        devstone_atomic<T>::internal_cycles = 100;
-    }
-};
-
+struct coupledHI_in_port : public cadmium::in_port<int>{};
+struct coupledHI_out_port : public cadmium::out_port<int>{};
 
-std::shared_ptr<cadmium::dynamic::modeling::coupled<TIME>> create_HI_model(uint width, uint depth) {
+std::shared_ptr<cadmium::dynamic::modeling::coupled<TIME>> create_HI_model(
+        uint width, uint depth, int ext_cycles, int int_cycles, TIME time_advance) {
```

Comment:
uint is not standard, I didn't check but it may fail in some compilers, unsigned int is safe to use.